src/ralph/api/routers/statements.py

```diff
@@ -254,3 +259,93 @@ async def get(
             statement["_source"] for statement in es_response["hits"]["hits"]
         ],
     }
+
+
+@router.post(
+    "/",
+    responses={
+        400: {
+            "model": ErrorDetail,
+            "description": "The request was invalid.",
+        },
+        409: {
+            "model": ErrorDetail,
+            "description": "Statements had a conflict with existing statements.",
+        },
+    },
+)
+# pylint: disable=unused-argument
+async def post(statements: Union[LaxStatement, List[LaxStatement]]):
+    """
+    Store a set of statements (or a single statement as a single member of a set).
+    NB: at this time, using POST to make a GET request, is not supported.
+
+    LRS Specification:
+    https://github.com/adlnet/xAPI-Spec/blob/1.0.3/xAPI-Communication.md#212-post-statements
+    """
+
+    # As we accept both a single statement as a dict, and multiple statements as a list,
+    # we need to normalize the data into a list in all cases before we can process it.
+    if not isinstance(statements, list):
+        statements = [statements]
+
+    # The statements dict has multiple functions:
+    # - generate IDs for statements that are missing them;
+    # - use the list of keys to perform validations and as a final return value;
+    # - provide an iterable containing both the statements and generated IDs for bulk.
+    statements_dict = {
+        str(statement.id) if statement.id else str(uuid4()): statement
+        for statement in statements
+    }
+
+    # Requests with duplicate statement IDs are considered invalid
+    # statements_ids were deduplicated by the dict, statements list was not
+    statements_ids = list(statements_dict.keys())
+    if len(statements) != len(statements_ids):
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Duplicate statement IDs in the list of statements",
+        )
+
+    # pylint: disable=unexpected-keyword-arg
+    es_response = ES_CLIENT.search(
+        index=ES_INDEX, body={"query": {"terms": {"_id": statements_ids}}}
```

Comment:
Does the following avoid having to disable the `unexpected-keyword-arg` pylint warning?
```suggestion
        index=ES_INDEX, query={"terms": {"_id": statements_ids}}
```

Reply:
`pylint` also complains about the `query` argument. I'm not sure if its a problem with `pylint` or `python-elasticsearch` but some arguments are just not getting picked up by `pylint`.